### DISCUSSION TITLE
OpenCV with CUDA

### DISCUSSION THREAD

**Discussion by Cheerpipe** [blakeblackshear/frigate#7730] [Ideas]
Hello all.

I just wonder if CPU usage may be reduced by using opencv cuda for tasks like motion detection and other ¿Heavy? frigate tasks currently using opencv with GPU.

Also, ¿What are the opencv most cpu consuming tasks? 

Thanks in advance and sorry for my crude English.

**Comment by NickM-27** [comment]
Frigate doesn't use opencv for object detection, and already supports using the Nvidia GPU for object detection. 

In 0.13 opencv has also been replaced by some other libraries in the motion pipeline to reduce cpu usage in motion detection.

**Comment by Cheerpipe** [comment]
Sorry, I mixed object with motion 

Object detection is already super fast and cpu friendly using a Coral tpu or compatible GPU. Just edited my original post.

**Reply by NickM-27** [reply]
Yeah, in 0.13 CPU usage for motion detection is reduced. It could potentially be run on the GPU but that hasn't been explored and may require refactoring elsewhere. 

I think the biggest issue would be ensuring similar behavior between motion detectors and respecting all of the same config parameters

**Reply by Cheerpipe** [reply]
If you were to try some gpu accelerated cv2 functions, where would you start?  I would like to do some test using cuda cv2 starting with motion code (complete new to python by the way) but don't know what are the most expensive cv2 calls in there.

**Reply by NickM-27** [reply]
I mean that depends on the config. As detect resolution increases so does the demand on motion detection. Best bet would be to use pyspy to see what opencv calls are taking the most cpu time

**Comment by Cheerpipe** [comment]
OpenCL may be a better alternative as cuda works only in Nvidia hardware while OpenCL OpenCV on AMD and Nvidia (doesn't know Intel).
